railties/lib/rails/engine.rb

```diff
@@ -615,6 +615,15 @@ def load_seed
       end
     end
 
+    initializer :add_fixture_paths do
+      next if is_a?(Rails::Application)
+
+      fixtures = config.root.join("test", "fixtures")
+      if fixtures.exist? && fixtures.to_s.include?(Rails.root.to_s)
```

Comment:
I'm not sure if this is the right check, but we need something here to filter out `test/fixtures` paths in engines for gems, etc otherwise their fixtures will potentially clobber the application's fixtures.